src/test/scala/com/databricks/spark/csv/CsvWriteSuite.scala

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2014 Databricks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.databricks.spark.csv
+
+import java.io.File
+
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.{FileSystem, FileUtil, Path}
+import org.apache.spark.SparkContext
+import org.apache.spark.sql.types._
+import org.apache.spark.sql.{DataFrame, SQLContext}
+import org.scalatest.{BeforeAndAfterAll, FunSuite}
+
+import scala.io.Source
+
+class CsvWriteSuite extends FunSuite with BeforeAndAfterAll {
+
+  val datesFile = "src/test/resources/dates.csv"
+  val tempEmptyDir = "target/test/empty/"
+
+  def parserLib: String = "COMMONS"
+
+  private var sqlContext: SQLContext = _
+
+  override protected def beforeAll(): Unit = {
+    super.beforeAll()
+    sqlContext = new SQLContext(new SparkContext("local[2]", "CsvWriteSuite"))
+  }
+
+  override protected def afterAll(): Unit = {
+    try {
+      sqlContext.sparkContext.stop()
+    } finally {
+      super.afterAll()
+    }
+  }
+
+  test("Save with default date format") {
+    mkTempDir()
+    val dates = readDatesFromFile()
+    val copyFilePath = tempEmptyDir + "dates-copy.csv"
+    val retDataFile = tempEmptyDir + "dates-result.csv"
+
+    // Write dataframe this way prior in spark 1.3 and before
+    dates.saveAsCsvFile(copyFilePath, Map("header" -> "false"))
+
+    // This method of writing can be used after spark 1.3
```

Comment:
Check out existing tests. You can use `assume` to ignore a specific test for older versions of Spark.